### ISSUE TITLE
Define a color model for output

### ISSUE THREAD

**Issue by jakecjones** [The-Lunar-Curve/SBC-POC#7]

**Comment by jakecjones** [comment]
Essentially we want users to use our tool to prep their image for saving. We will output an array of dominant or selected colors. Downstream this will help a user search for images by a color or color palettes. In the future it could be used by a person to take a picture and find matching color schemes too. The task I outlined is to determine what the dominant color model includes while thinking through how a user or automated system would search for similar colors.

**Comment by jakecjones** [comment]
See [documentation](https://github.com/The-Lunar-Curve/SBC-POC/wiki/Color-Extraction) for current model

**Comment by ordinarybrian** [comment]
Since this is specifically a digital, web tool, I think it would be fine to base the color model on RGB. It's very easy to convert RGB to hex, a little less so to convert RGB to HSB, but still doable.

Since it's all based on pixel color, we don't have to worry about alphas or anything like that either. I recommend RGB as the standard.

**Comment by jakecjones** [comment]
@bgcbrianclark I think we should assume our user doesn't know how to run the conversion when considering outputs. I agree on the alpha channel although that channel is available if we need it.

**Comment by ordinarybrian** [comment]
Well we can handle all of the conversion. From what I understand, user uploads photo > system pulls dominant colors > system outputs a selection of different colors the user can pick > user picks a color and receives color info. Is that correct? Essentially, we can store everything RGB and when the user chooses a color we just put up the RGB, HSL and Hex conversions for them to use.

Am I on the right track?

**Comment by jakecjones** [comment]
Your understanding is that we convert when a user selects a color from the array but that's just one use case. The other use case is once the color extract event finishes, the user may want to save all of the dominant colors to a database. They may not want to copy any color codes or need to read that data directly. The desired output here is an array of objects, each object signifying all the data we can extract on one color.

**Comment by ordinarybrian** [comment]
Right, so we'd want a master color model, which would get converted to multiple outputs whenever it's called. We could store RGB, and convert whenever the user needs to reference, or we could convert and store the HSL and hex in the DB? I think we should do the conversions on the front end, but that's probably because I'm primarily a front-end guy. What do you suggest?

**Comment by jakecjones** [comment]
All conversions happen on client if that helps you understand.
